eslint/eslint-patch/src/eslint-bulk-suppressions/bulk-suppressions-patch.ts

```diff
@@ -15,265 +14,277 @@ interface ISuppression {
+    serializedSuppression: string;
+  };
+}
 
-  return null;
+function getNodeName(node: TSESTree.Node): string | undefined {
+  if (Guards.isClassDeclarationWithName(node)) {
+    return node.id.name;
+  } else if (Guards.isFunctionDeclarationWithName(node)) {
+    return node.id.name;
+  } else if (Guards.isClassExpressionWithName(node)) {
+    return node.id.name;
+  } else if (Guards.isFunctionExpressionWithName(node)) {
+    return node.id.name;
+  } else if (Guards.isNormalVariableDeclaratorWithAnonymousExpressionAssigned(node)) {
+    return node.id.name;
+  } else if (Guards.isNormalObjectPropertyWithAnonymousExpressionAssigned(node)) {
+    return node.key.name;
+  } else if (Guards.isNormalClassPropertyDefinitionWithAnonymousExpressionAssigned(node)) {
+    return node.key.name;
+  } else if (Guards.isNormalAssignmentPatternWithAnonymousExpressionAssigned(node)) {
+    return node.left.name;
+  } else if (Guards.isNormalMethodDefinition(node)) {
+    return node.key.name;
+  } else if (Guards.isTSEnumDeclaration(node)) {
+    return node.id.name;
+  } else if (Guards.isTSInterfaceDeclaration(node)) {
+    return node.id.name;
+  } else if (Guards.isTSTypeAliasDeclaration(node)) {
+    return node.id.name;
+  }
 }
 
 type NodeWithParent = TSESTree.Node & { parent?: TSESTree.Node };
 
 function calculateScopeId(node: NodeWithParent | undefined): string {
   const scopeIds: string[] = [];
   for (let current: NodeWithParent | undefined = node; current; current = current.parent) {
-    const scopeIdForASTNode: string | null = getNodeName(current);
-    if (scopeIdForASTNode !== null) scopeIds.unshift(scopeIdForASTNode);
+    const scopeIdForASTNode: string | undefined = getNodeName(current);
+    if (scopeIdForASTNode !== undefined) {
+      scopeIds.unshift(scopeIdForASTNode);
+    }
   }
 
-  if (scopeIds.length === 0) return '.';
-
-  return '.' + scopeIds.join('.');
+  if (scopeIds.length === 0) {
+    return '.';
+  } else {
+    return '.' + scopeIds.join('.');
+  }
 }
 
-/**
- * @throws Throws an error if the command to retrieve the root path fails.
- * @returns The root path of the monorepo.
- */
-export function getGitRootPath(): string {
-  const result: SpawnSyncReturns<string> = spawnSync('git', ['rev-parse', '--show-toplevel'], {
-    encoding: 'utf-8'
-  });
-  if (result.status !== 0) throw new Error(`get root path failed`);
-  return result.stdout.toString().trim();
-}
+const eslintrcPathByFileOrFolderPath: Map<string, string> = new Map();
 
-export const GitRootPath: string = getGitRootPath();
+function findEslintrcFolderPathForNormalizedFileAbsolutePath(normalizedFilePath: string): string {
+  const cachedFolderPathForFilePath: string | undefined =
+    eslintrcPathByFileOrFolderPath.get(normalizedFilePath);
+  if (cachedFolderPathForFilePath) {
+    return cachedFolderPathForFilePath;
+  }
+  const normalizedFileFolderPath: string = normalizedFilePath.substring(
+    0,
+    normalizedFilePath.lastIndexOf('/')
```

Comment:
I was always under the impression the advantage of path.dirname() is that it is agnostic to Windows filesystems that sometimes use backslash '\'?

Reply:
It does, but with this change I'm normalizing the path names to always use forward slashes.